plugins/by-name/langmapper/default.nix

```
@@ -0,0 +1,104 @@
+{
+  lib,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkOption
+    optional
+    types
+    mkIf
+    concatStringsSep
+    map
+    ;
+  inherit (lib.nixvim.lua) toLua;
+  inherit (lib.nixvim.defaultNullOpts) mkBool mkListOf;
+
+  mkKeyModes = mkListOf lib.nixvim.keymaps.modeType;
+  cfg = config.plugins.langmapper;
+in
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "langmapper";
+  package = "langmapper-nvim";
+  description = "A plugin that makes Neovim more friendly to non-English input methods";
+  maintainers = [ lib.maintainers.shved ];
+  settingsOptions = {
+    map_all_ctrl = mkBool true "Add mapping for every CTRL+ binding or not";
+    ctrl_map_modes = mkKeyModes [
+      "n"
+      "o"
+      "i"
+      "c"
+      "t"
+      "v"
+    ] "Modes to `map_all_ctrl`";
```

This might format nicer if we use [`lib.stringToCharacters`](https://noogle.dev/f/lib/stringToCharacters):

```suggestion
    ctrl_map_modes = mkKeyModes (lib.stringToCharacters "noictv") ''
      Modes to `map_all_ctrl`
    '';
```